TOMM20.py

```diff
@@ -0,0 +1,263 @@
+    """
+    Abstract class for working with biological sequences
+    Attributes of BiologicalSequence class:
+    -   seq (str): biological sequence
+    -   seq_type (str): which sequence DNA, RNA or protein the sequence is
+    """
+
+    def __init__(self, seq):
+        self.seq = seq
+        self.seq_type = None
+
+    def __len__(self):
+        return len(self.seq)
+
+    def __getitem__(self, idx):
+        return self.seq[idx]
+
+    def __getslice__(self, start, end):
+        return self.seq[start: end]
+
+    def __repr__(self):
+        return f'The sequence is: {self.seq}, type is {self.seq_type}'
+
+    def __str__(self):
+        return self.seq
+
+    def check_seq_type(self, check_type):
+        """
+        Check if sequence is check_type of molecule DNA, RNA or protein
+        Input:
+        -   check_type (str): type of molecule DNA, RNA, protein
+        Return:
+        -   bool: True is type match with check_type, False if not
+        """
+
+        seq_types = {
+            'DNA': ALPHABET_FOR_DNA,
+            'RNA': ALPHABET_FOR_RNA,
+            'Protein': ALPHABET_FOR_PROTEIN
+        }
+
+        if check_type not in seq_types:
+            raise ValueError(f'There is not such sequence type as {check_type}!')
+        return set(self.seq) <= seq_types[check_type]
+
+    def type_definition(self):
+        """
+        Set sequence type (DNA, RNA, protein).
+        Input:
+        -   seq (str): sequence that we need to get type:
+        Return:
+        -   type (str): type that sequence are
+        """
+        if self.check_seq_type('DNA'):
+            self.seq_type = 'DNA'
+        elif self.check_seq_type('RNA'):
+            self.seq_type = 'RNA'
+        elif self.check_seq_type('Protein'):
+            self.seq_type = 'Protein'
+        else:
+            raise ValueError(f'Sequence {self.seq} can not be analysed!')
+
+
+class NucleicAcidSequence(BiologicalSequence):
+    """
+    Methods for nucleic acid sequences: DNA & RNA
+    Attributes:
+    -   seq (str): nucleic acid sequence
+    -   seq_type (str): type of nucleic acid sequence (DNA or RNA)
+    -   complement_alphabet (dict): dictionary with complement pairs for DNA or RNA
+    """
+
+    def __init__(self, seq):
+        super().__init__(seq)
+        super().type_definition()
+        if super().check_seq_type('DNA'):
+            self.complement_alphabet = COMPLEMENT_ALPHABET_DNA
+        elif super().check_seq_type('RNA'):
+
+            self.complement_alphabet = COMPLEMENT_ALPHABET_RNA
+        else:
+            raise ValueError(f'Sequence {self.seq} is not nucleic acid!')
+
+    def complement(self):
+        """
+        Complement DNA or RNA sequences to RNA or DNA sequence
+        Return:
+        -   complement sequence
+        """
+        output_seq = []
+        for nucleotide in self.seq:
+            output_seq.append(self.complement_alphabet[nucleotide])
+        return NucleicAcidSequence(''.join(output_seq))
```

Comment:
функция map() должна работать чуть побыстрее
```suggestion
output_seq = ''.join(map(lambda nucleotide: self.complement_alphabet[nucleotide], self.seq))
return NucleicAcidSequence(output_seq)
```

Reply:
Ну вот мы только сейчас прошли map и lambda :)
Но да, так более функционально